lib/oneboxer.rb

```diff
@@ -397,9 +397,14 @@ def self.external_onebox(url, available_strategies = nil)
       available_strategies ||= Oneboxer.ordered_strategies(uri.hostname)
       strategy = available_strategies.shift
 
-      fd = FinalDestination.new(url, get_final_destination_options(url, strategy))
+      fd = FinalDestination.new(
+        url,
+        get_final_destination_options(url, strategy).merge(stop_at_blocked_pages: true)
```

Comment:
I'm not sure if you saw it, but https://github.com/discourse/discourse/pull/15689 is sort of the inverse of this. I'm wondering if we should revert that PR. (You can look the tests in that PR for a better description of what it does)

Reply:
Yep, saw the PR and the specs you added started failing after I made this change so I updated them. :+1: